File: Examples/Perl/extracting-managing-ocr/without-cloud-storage/extract-ocr-or-hocr-text-from-image-without-storage.pl
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeOcrCloud::OcrApi;
use AsposeOcrCloud::ApiClient;
use AsposeOcrCloud::Configuration;

my $configFile = '../../data/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);

my $data_path = '../../data/';
my $out_path = $configProps->{'out_folder'};;

#ExStart:1
$AsposeOcrCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeOcrCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeOcrCloud::Configuration::debug = 1;

#Instantiate Aspose.OCR API SDK
my $ocrApi = AsposeOcrCloud::OcrApi->new();

#Set the image file name
my $name = 'Sampleocr.bmp';

#Set the language of the document
my $language = 'English';

#Set the spelling correction is used
my $useDefaultDictionaries = 'true';

#invoke Aspose.OCR Cloud SDK API to extract image text from local file                           
my $response = $ocrApi->PostOcrFromUrlOrContent(language=> $language, useDefaultDictionaries=>$useDefaultDictionaries, file => $data_path.$name);

if($response->{'Status'} eq 'OK'){
    print "\n Text :: " . $response->{'Text'};
    
    print "\n Extract OCR or HOCR Text from Images without using Storage, Done!";
	
}
#ExEnd:1

File: Examples/PHP/extracting-managing-ocr/without-cloud-storage/extract-ocr-or-hocr-text-from-image-without-storage.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/../..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/../..') . '/Utils.php';

use Aspose\OCR\OcrApi;
use Aspose\OCR\AsposeApp;

class RecognizeText {

	public $barcodeApi;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->ocrApi = new OcrApi();
	}

	public function recognizeImageTextFromTheRequestBody() {

		//Set the image file name
		$name = "Sampleocr.bmp";

		//Set the image file url (if any)
		$url = null;

		//Set the language of the document
		$language = "English";

		//Set the spelling correction is used
		$useDefaultDictionaries = true;

		// Set local file (if any)
		$file = realpath(__DIR__ . '/../../../..') . '/Data/' . $name;

		//invoke Aspose.OCR Cloud SDK API to extract image text from local file
		$response = $this->ocrApi->PostOcrFromUrlOrContent($url, $language, $useDefaultDictionaries, $file);
		print_r ( $response );
	}
}

$recognizeText = new RecognizeText();
$recognizeText->recognizeImageTextFromTheRequestBody();

?>
//ExEnd:
